implementations/rust/ockam/ockam_command/src/vault/mod.rs

```diff
@@ -80,6 +84,27 @@ async fn run_impl(ctx: Context, (opts, cmd): (CommandGlobalOpts, VaultCommand))
             opts.state.identities.create(&idt_name, idt_config)?;
             println!("Identity attached to vault: {}", idt_name);
         }
+        VaultSubcommand::Show { name } => {
+            let state = match name {
+                Some(name) => opts.state.vaults.get(&name),
+                None => opts.state.vaults.default(),
+            }?;
```

Comment:
This could be done automatically if you define the `name` argument as described [here](https://github.com/build-trust/ockam/issues/3931#issue-1499868554), but for the identity in this case.

We will probably have more than one identity command with this argument, and we can reuse this piece of code by configuring the command's argument properly.

Reply:
I'm reopening this again because I think I didn't explain myself properly. [Here is an example](https://github.com/build-trust/ockam/blob/5ad7c89df71230b2ea4016f08d6f08fd2042245f/implementations/rust/ockam/ockam_command/src/service/start.rs#L26) of what I meant.

In your case, the `default_vault_name` will return an `Option<String>` and will contain what I highlighted above.

Let me know if it's clear now!

Reply:
Isn't `default_vault_name` return an `String`? And  `vault` is the default vault name because `default_path` is `CliState::defaults_dir()?.join("vault")`?

```rust
// ockam/implementations/rust/ockam/ockam_api/src/cli_state.rs:148
pub fn default_path(&self) -> Result<PathBuf> {
    Ok(CliState::defaults_dir()?.join("vault"))
} 
```

Reply:
Yeah, you are right, and also we can't return `Result` on clap's default methods. We can instead do a little improvement and write that part as follows:

```rust
let name = name.unwrap_or(file_stem(&opts.state.vaults.default()?.path)?);
let state = opts.state.vaults.get(&name)?;
```

Or even better, you could add a new method to the `VaultState` to retrieve the vault name, which would be a function returning the output of `file_stem(self.path)` so we can write the above as:

```rust
let name = name.unwrap_or(&opts.state.vaults.default()?.name()?);
let state = opts.state.vaults.get(&name)?;
```